cloud/cluster/memory/cluster.go

```diff
@@ -0,0 +1,108 @@
+package memory
+
+import (
+	"github.com/scootdev/scoot/cloud/cluster"
+)
+
+func NewCluster(initial []cluster.Node, updateCh chan []cluster.NodeUpdate, wrapper cluster.NodeWrapper) cluster.Cluster {
+	if wrapper != nil {
+		for i, member := range initial {
+			initial[i] = wrapper.Wrap(member)
+		}
+	}
+
+	c := &simpleCluster{
+		inCh:    updateCh,
+		reqCh:   make(chan interface{}),
+		wrapper: wrapper,
+		members: State{members: initial},
+		subs:    nil,
+	}
+	go c.loop()
+	return c
+}
+
+type simpleCluster struct {
+	inCh  chan []cluster.NodeUpdate
+	reqCh chan interface{}
+
+	wrapper cluster.NodeWrapper
+	members State
+	subs    []chan []cluster.NodeUpdate
+}
+
+func (c *simpleCluster) Members() ([]cluster.Node, error) {
+	ch := make(chan []cluster.Node)
+	c.reqCh <- ch
+	return <-ch, nil
+}
+
+func (c *simpleCluster) Subscribe() (cluster.Subscription, error) {
+	ch := make(chan cluster.Subscription)
+	c.reqCh <- ch
+	return <-ch, nil
+}
+
+func (c *simpleCluster) Close() error {
+	close(c.reqCh)
+	return nil
+}
+
+func (c *simpleCluster) done() bool {
+	return c.inCh == nil && c.reqCh == nil
+}
+
+func (c *simpleCluster) loop() {
+	for !c.done() {
+		select {
+		case updates, ok := <-c.inCh:
+			for i, _ := range updates {
+				if updates[i].Node != nil {
+					updates[i].Node = c.wrapper.Wrap(updates[i].Node)
+				}
+			}
+			if !ok {
+				c.inCh = nil
+				continue
+			}
+			c.members.Apply(updates)
+			for _, sub := range c.subs {
+				sub <- updates
```

Comment:
This would be in addition to the subscription.go:loop(). Let me put it this way, it's unlikely that what I'm suggesting is actually necessary, it falls under the category of 'it signals intent' in the case where there are multiple listeners and we're not guaranteed that those listeners are implemented a specific way (we could wrap each sub channel in a type...)